mutex.h: No such file or directory

Fail to compile current master branch.

File: src/epoll.hpp:44:19 references mutex.h, but there is no such file. File name that exists is mutex.hpp.

I've updated epoll.hpp file and it works.

See full stack:

$ make
Making all in doc
make[1]: Entering directory '/devel/libzmq/doc'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/devel/libzmq/doc'
make[1]: Entering directory '/devel/libzmq'
  CXX      src/src_libzmq_la-address.lo
  CXX      src/src_libzmq_la-client.lo
In file included from src/poller.hpp:42:0,
                 from src/socket_base.hpp:41,
                 from src/client.hpp:33,
                 from src/client.cpp:32:
src/epoll.hpp:44:19: fatal error: mutex.h: No such file or directory
 #include "mutex.h"
                   ^
compilation terminated.
make[1]: *** [Makefile:3274: src/src_libzmq_la-client.lo] Error 1
make[1]: Leaving directory '/devel/libzmq'
make: *** [Makefile:4035: all-recursive] Error 1

--- Archlinux on RaspberryPI ---

--- a/src/epoll.cpp
+++ b/src/epoll.cpp
@@ -92,9 +92,7 @@ void zmq::epoll_t::rm_fd (handle_t handle_)
     int rc = epoll_ctl (epoll_fd, EPOLL_CTL_DEL, pe->fd, &pe->ev);
     errno_assert (rc != -1);
     pe->fd = retired_fd;
-    retired_sync.lock ();
     retired.push_back (pe);
-    retired_sync.unlock ();
 
     //  Decrease the load metric of the thread.
     adjust_load (-1);
@@ -182,12 +180,10 @@ void zmq::epoll_t::loop ()
         }
 
         //  Destroy retired event sources.
-        retired_sync.lock ();
         for (retired_t::iterator it = retired.begin (); it != retired.end (); ++it) {
             LIBZMQ_DELETE(*it);
         }
         retired.clear ();
-        retired_sync.unlock ();
     }
 }
 

--- a/src/epoll.hpp
+++ b/src/epoll.hpp
@@ -41,7 +41,6 @@
 #include "fd.hpp"
 #include "thread.hpp"
 #include "poller_base.hpp"
-#include "mutex.h"
 
 namespace zmq
 {
@@ -102,9 +101,6 @@ namespace zmq
 
         //  Handle of the physical thread doing the I/O work.
         thread_t worker;
-
-        // Synchronisation of the retired event source
-        mutex_t retired_sync;
 
         epoll_t (const epoll_t&);
         const epoll_t &operator = (const epoll_t&);

